src/Microsoft.Diagnostics.EventFlow.Inputs.ILogger/Logger/EventFlowLogger.cs

```diff
@@ -0,0 +1,111 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Internal;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs.ILogger.Logger
+{
+    internal class EventFlowLogger : Extensions.Logging.ILogger
+    {
+        private readonly string categoryName;
+        private readonly LoggerInput loggerInput;
+        private Scope<object> scope;
+
+        public EventFlowLogger(string categoryName, LoggerInput loggerInput)
+        {
+            Debug.Assert(categoryName != null);
+            Debug.Assert(loggerInput != null);
+
+            this.categoryName = categoryName;
+            this.loggerInput = loggerInput;
+        }
+
+        public void Log<TState>(Extensions.Logging.LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+                return;
+            var message = formatter(state, exception);
+
+            Dictionary<string, object> properties = new Dictionary<string, object>();
+            if (state is FormattedLogValues)
+            {
+                var formattedState = state as FormattedLogValues;
+                //last KV is the whole message, we will pass it separately
+                for (int i = 0; i < formattedState.Count - 1; i++)
+                    properties.Add(formattedState[i].Key, formattedState[i].Value);
+            }
+            if (scope?.State != null)
+            {
+                var formattedState = scope.State as FormattedLogValues;
+                if (formattedState != null)
+                {
+                    for (int i = 0; i < formattedState.Count - 1; i++)
+                    {
+                        if (!properties.ContainsKey(formattedState[i].Key)) //message and scope may have the same property
+                            properties.Add(formattedState[i].Key, formattedState[i].Value);
+                    }
+                    //last KV is the whole 'scope' message, we will add it formatted
+                    properties.Add("Scope", formattedState.ToString());
```

Comment:
Agreed that all inputs should behave the same way. Feel free to open an issue about this and we can deal with it as a separate change.

I think whatever the original event carries should take priority over what the input adds. So if there is a "Message" property on the event, we should not overwrite it. We could either skip input additions completely or make the added property name unique, e.g. by appending random number at the end of the name. These are, essentially, variations of the (1) option you have. I have slight preference for the latter. 

I do not think using a separate dictionary is a good idea. It will make the output code and, ultimately, consumption of the data much more complicated, for a marginal benefit.

Oh, and I would just report a warning, not an error, if a name collision happens, esp. if you decide to make the added property name unique and include the data, mitigating the conflict.